test/unix/main.ml

```diff
@@ -623,11 +623,151 @@ end
 
 (* Tests of {Index.close} *)
 module Close = struct
+  exception Stop
+
+  let check_logs msg rw n_log n_log_async =
+    let log = Index.log rw in
+    let log_async = Index.log_async rw in
+    let len = Option.map List.length in
+    Alcotest.(check (option int)) (msg ^ ": log entries") n_log (len log);
+    Alcotest.(check (option int))
+      (msg ^ ": log_async entries")
+      n_log_async (len log_async)
+
+  let force_merge rw =
+    let thread = Index.try_merge_aux ~force:true rw in
+    Index.await thread |> check_completed;
+    check_logs "force_merge" rw (Some 0) None
+
   let close_reopen_rw () =
-    let* Context.{ rw; tbl; clone; _ } = Context.with_full_index () in
-    Index.close rw;
-    let w = clone ~readonly:false () in
-    check_equivalence w tbl
```

Comment:
Perhaps there's value in keeping a shorter test that doesn't trigger a crash? (And adding the other ones too.)

Reply:
I've re-added this simple test